For #8050 feat(nimbus): Add Audience page to sidebar

eliserichards opened the pull request:
📢 Requires #8087 and #8086 to be merged first

Because...
* We want to reuse the Audience page to allow users to edit rollouts while live

This commit...
* Adds the Audience page to the sidebar **for both rollouts and experiments**
* **Hides** the audience link for experiments, **enables** it for rollouts
* Because of #8096, the fields on the audience page are **disabled** 
   * we will enable them once we implement the ability to send the edits through review - #7823

---- 

Rollout with fully disabled audience page:
<img width="1792" alt="image" src="https://user-images.githubusercontent.com/43795363/211891701-c91d1745-4dbc-4f09-b668-b08baa602c0f.png">

Experiment with no audience link in sidebar: 
   
<img width="1507" alt="image" src="https://user-images.githubusercontent.com/43795363/212999545-616e580f-87de-4563-b8a0-09b95fa7c77f.png">

<img width="248" alt="image" src="https://user-images.githubusercontent.com/43795363/212999435-6cd25723-2b93-488e-8e67-b8a4d07c9b66.png">

eliserichards commented:
Actually, I'm gonna close this for now until the other ones get merged 🤞

jaredlockhart reviewed (dismissed):
Looks good, where's tests? 🤔

yashika-khurana commented:
> Looks good, where's tests? 🤔

Interesting and code coverage is 100% 🤔

yashika-khurana commented:
what about not showing the Audience option at all for the experiments instead of disabling it (like that we have now) and only showing the Audience option for the rollouts? @jaredlockhart @eliserichards

jaredlockhart commented:
> what about not showing the Audience option at all for the experiments instead of disabling it (like that we have now) and only showing the Audience option for the rollouts? @jaredlockhart @eliserichards

Yeah I think that's a good option too, since experiments should never be able to get to it.

eliserichards commented:
Makes sense to me. I wasn't sure if we'd want to see Audience from an experiment in the future, but I changed it so it's only shown for rollouts now 😄 I also added some tests. It's kinda weird that code cov wasn't complaining 🤔

yashika-khurana reviewed (approved):
tested locally works awesome, audience option is gone for experimenter, thank you for adding test as well, 🎉
